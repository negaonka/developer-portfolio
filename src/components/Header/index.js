import React from 'react'
import Nav from '../Nav';

function Header(props){
    const {
        menus = [],
        setCurrentMenu,
        currentMenu
      } = props;
    return(
    <>
    <header className="flex-row px-1 py-2">
      <h2>
        <a data-testid="link" href="/">
          <span role="img" aria-label="name"> Neha Gaonkar</span>
        </a>
      </h2>
    <Nav   
        menus={menus}
        setCurrentMenu={setCurrentMenu}
        currentMenu={currentMenu}/>
    </header>
    </>
    )
}

export default Header;