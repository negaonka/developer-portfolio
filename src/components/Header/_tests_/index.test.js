import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '..';

const menus = [
    { name: 'about me' }
  ]
  const mockCurrentMenu = jest.fn();
  const mockSetCurrentMenu = jest.fn();

afterEach(cleanup);

describe('Header component', () => {
    // baseline test
    it('renders', () => {
        render(<Header 
            menus={menus}
            setCurrentMenu={mockSetCurrentMenu}
            currentMenu={mockCurrentMenu}/>);
      });

    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Header   
            menus={menus}
            setCurrentMenu={mockSetCurrentMenu}
            currentMenu={mockCurrentMenu}/>);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
      });
    })

    //link visibility
    describe('links are visible', () => {
        it('inserts text into the links', () => {
          // Arrange
          const { getByTestId } = render(<Header   
            menus={menus}
            setCurrentMenu={mockSetCurrentMenu}
            currentMenu={mockCurrentMenu}/>);
          // Assert
          expect(getByTestId('link')).toHaveTextContent('Neha Gaonkar');
        });
      })