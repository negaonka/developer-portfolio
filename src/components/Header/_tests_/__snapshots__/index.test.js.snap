// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1 py-2"
  >
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="name"
          role="img"
        >
          Neha Gaonkar
        </span>
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2 false"
        >
          <span>
            About me
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
